app/src/test/java/com/example/mylittlelibrary/BookUnitTest.kt

```diff
@@ -0,0 +1,86 @@
+package com.example.mylittlelibrary
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.Observer
+import com.example.mylittlelibrary.data.Book
+import com.example.mylittlelibrary.service.BookService
+import com.example.mylittlelibrary.ui.viewModel.BookViewModel
+import io.mockk.impl.annotations.MockK
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.*
+import org.junit.rules.TestRule
+import kotlinx.coroutines.newSingleThreadContext
+import org.junit.After
+
+import org.junit.Before
+
+class BookUnitTest {
```

Comment:
Added IBookSerive Inteface to prevent concretion and follow SOLID principle.
https://www.digitalocean.com/community/conceptual_articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design

Reply:
Really nice unit test.